src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/SurveySvcRestClientImpl.java

```diff
@@ -123,46 +125,54 @@ private SurveyDTO getSurveyDtoFromResponseEntity(ResponseEntity<String> response
   }
 
   @Override
-  public SurveyDTO findSurvey(UUID surveyId) throws RestClientException {
+  public SurveyDTO findSurvey(final UUID surveyId) throws RestClientException {
     UriComponents uriComponents =
         restUtility.createUriComponents(
             appConfig.getSurveySvc().getSurveyDetailPath(), null, surveyId);
 
     HttpEntity<?> httpEntity = restUtility.createHttpEntity(null);
+    ResponseEntity<String> responseEntity = null;
+    SurveyDTO survey = null;
 
     try {
       log.debug(
           "about to get to the Survey SVC with surveyId {} from {}",
           surveyId,
           uriComponents.toUri());
-      ResponseEntity<String> responseEntity =
+      responseEntity =
           restTemplate.exchange(uriComponents.toUri(), HttpMethod.GET, httpEntity, String.class);
-
-      return getSurveyDtoFromResponseEntity(responseEntity);
-    } catch (RestClientException e) {
-      return null;
+      survey = getSurveyDtoFromResponseEntity(responseEntity);
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+        return null;
+      }
+      log.error("Client error: ", e);
     }
+    return survey;
   }
 
   @Override
-  public SurveyDTO findSurveyByRef(String surveyRef) throws RestClientException {
+  public SurveyDTO findSurveyByRef(final String surveyRef) throws RestClientException {
     UriComponents uriComponents =
         restUtility.createUriComponents(
-            appConfig.getSurveySvc().getSurveyRefPath(), null, surveyRef);
+            appConfig.getSurveySvc().getSurveyRefPath(),
+            null, surveyRef);
 
     HttpEntity<?> httpEntity = restUtility.createHttpEntity(null);
+    ResponseEntity<String> responseEntity = null;
+    SurveyDTO survey = null;
 
     try {
-      log.debug(
-          "about to get to the Survey SVC with surveyRef {} from {}",
-          surveyRef,
-          uriComponents.toUri());
-      ResponseEntity<String> responseEntity =
+      responseEntity =
           restTemplate.exchange(uriComponents.toUri(), HttpMethod.GET, httpEntity, String.class);
 
-      return getSurveyDtoFromResponseEntity(responseEntity);
-    } catch (RestClientException e) {
-      return null;
+      survey =  getSurveyDtoFromResponseEntity(responseEntity);
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+        return null;
+      }
+      log.error("Client error: ", e);
```

Comment:
Again, rethrow the exception after logging it

Reply:
Maybe you could add some context via the status code here log.error("Client error statuscode={}", e.getStatusCode(), e);